### PR TITLE
Feature/sg 107 repeated agumentation sampler

### PR DESCRIPTION
**Pull request by shaydeci**
Repeated augmentation sampler for large batch training.

### CODE REVIEW

**Comment by lotem-deci** [comment]
@shaydeci  - Where are the tests?

**Comment by shaydeci** [comment]
> @shaydeci - Where are the tests?

Ill add a TODO comment as this sampler only supports DDP...

File: src/super_gradients/training/datasets/samplers/repeated_augmentation_sampler.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+# TODO: Add unit test for RepeatAugSampler once DDP unit tests are supported.
+
+class RepeatAugSampler(Sampler):
+    """
+    Sampler that restricts data loading to a subset of the dataset for distributed,
+    with repeated augmentation.
+    It ensures that different each augmented version of a sample will be visible to a
+    different process (GPU). Heavily based on torch.utils.data.DistributedSampler
+    This sampler was taken from https://github.com/facebookresearch/deit/blob/0c4b8f60/samplers.py
+    Copyright (c) 2015-present, Facebook, Inc.
```
Comment (ofrimasad) [inline src/super_gradients/training/datasets/samplers/repeated_augmentation_sampler.py:16]: add here "( Apache 2.0 License)"

**Review by ofrimasad — COMMENTED** [review]
few minor comments inline

**Pull request merged**